webpack.config.js

```diff
@@ -25,20 +23,23 @@ const rules = [{
 }]
 
 
-module.exports = {
-    entry: './src/index.tsx',
-    mode: production ? 'production' : 'development',
-    devtool: production ? undefined : 'source-map',
-    module: {
-        rules: rules
-    },
-    resolve: { extensions: ['*', '.js', '.jsx', '.ts', '.tsx'] },
-    output: {
-        filename: 'bundle.js'
-    },
-    devServer: {
-        port: 3000,
-        hot: true
-    },
-    plugins: [new webpack.HotModuleReplacementPlugin()]
+module.exports = (env) => {
+    const production = env.production;
+    return {
+        entry: './src/index.tsx',
+        mode: production ? 'production' : 'development',
+        devtool: production ? undefined : 'source-map',
+        module: {
+            rules: rules
+        },
+        resolve: { extensions: ['*', '.js', '.jsx', '.ts', '.tsx'] },
+        output: {
+            filename: 'bundle.js'
+        },
+        devServer: {
+            port: 3000,
```

Comment:
For the future I suggest to move this kind of config in a dedicated .env file for an easier management between different potential environments